fastlane/lib/fastlane/actions/carthage.rb

```diff
@@ -13,6 +13,7 @@ def self.run(params)
         cmd << "--verbose" if params[:verbose] == true
         cmd << "--platform #{params[:platform]}" if params[:platform]
         cmd << "--configuration #{params[:configuration]}" if params[:configuration]
+        cmd << "--derived-data '#{params[:derived_data]}'" if params[:derived_data]
```

Comment:
Please use `shellescape` instead of `'`. Make sure to actually remove the `'` surrounding the `derived_data`